jerry-core/jrt/jrt.h

```diff
@@ -153,4 +153,36 @@ void JERRY_ATTR_NORETURN jerry_fatal (jerry_fatal_code_t code);
 #define JERRY__LOG2_8(n) (((n) >= 1 << 8) ? (8 + JERRY__LOG2_4 ((n) >> 8)) : JERRY__LOG2_4 (n))
 #define JERRY_LOG2(n) (((n) >= 1 << 16) ? (16 + JERRY__LOG2_8 ((n) >> 16)) : JERRY__LOG2_8 (n))
 
+/**
+ * JERRY_BLOCK_TAIL_CALL_OPTIMIZATION
+ *
+ * Adapted from abseil ( https://github.com/abseil/ )
+ *
+ * Instructs the compiler to avoid optimizing tail-call recursion. This macro is
+ * useful when you wish to preserve the existing function order within a stack
+ * trace for logging, debugging, or profiling purposes.
+ *
+ * Example:
+ *
+ *   int f() {
+ *     int result = g();
+ *     JERRY_BLOCK_TAIL_CALL_OPTIMIZATION();
+ *     return result;
+ *   }
+ *
+ * This macro is intentionally here as jerryscript-compiler.h is a public header and
+ * it does not make sense to expose this macro to the public.
+ */
+#if defined (__clang__) || defined (__GNUC__)
+/* Clang/GCC will not tail call given inline volatile assembly. */
+#define JERRY_BLOCK_TAIL_CALL_OPTIMIZATION() __asm__ __volatile__ ("")
```

Comment:
These things seems random, which is hard to maintain. What about doing something before return, e.g clearing a status bit in the jerry context?

Reply:
That's would not stop tail optimization

Reply:
> These things seems random, which is hard to maintain. What about doing something before return, e.g clearing a status bit in the jerry context?

That's would not stop tail optimization

Reply:
Setting status bit seems to be working.

Reply:
> Setting status bit seems to be working.

Then would be good